rp2040-hal/src/timer.rs

```diff
@@ -14,10 +14,26 @@ use crate::atomic_register_access::{write_bitmask_clear, write_bitmask_set};
 use crate::pac::{RESETS, TIMER};
 use crate::resets::SubsystemReset;
 use core::marker::PhantomData;
+use core::sync::atomic::{AtomicU8, Ordering};
 
 /// Instant type used by the Timer & Alarm methods.
 pub type Instant = TimerInstantU64<1_000_000>;
 
+static ALARMS: AtomicU8 = AtomicU8::new(0x0F);
+fn take_alarm(mask: u8) -> bool {
+    critical_section::with(|_| {
+        let alarms = ALARMS.load(Ordering::Relaxed);
+        ALARMS.store(alarms & !mask, Ordering::Relaxed);
+        (alarms & mask) != 0
```

Comment:
I think this should check that only one bit in `mask` is set. (Or, easier, don't pass a mask to this function, but the bit number).
Otherwise one could call it with multiple bits set, which would return `true` in case one of the alarms was still available, but the caller wouldn't know which one.
Of course this is not critical as this function isn't `pub`.

Reply:
This is an internal method so I figured it'd be less change to reuse the `$armed_bit_mask` already defined in the macro.